Sources/FDB/Transaction/Transaction+NIO.swift

```diff
@@ -394,7 +408,12 @@ public extension FDB.Transaction {
         } catch {
             promise.fail(error)
         }
-
-        return promise.futureResult
+        
+        switch shouldCommit {
+        case true:
+            return self.commit().flatMap { promise.futureResult }
+        case false:
+            return promise.futureResult
+        }
```

Comment:
Do you actually need a `switch` for a simple comparison? :D I would do something like this or simple `if/else`
```suggestion
        return shouldCommit
            ? self.commit().flatMap { promise.futureResult }
            : promise.futureResult
```